bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,309 @@
+            return first_kata_id
+
+    async def kata_information(self, kata_id: str) -> dict:
+        """
+        Returns the information about the Kata.
+
+        Uses the codewars.com API to get information about the kata using the kata's ID.
+        """
+        async with self.bot.http_session.get(API_ROOT.format(kata_id=kata_id)) as response:
+            if response.status != 200:
+                logger.error(
+                    f"Unexpected status code {response.status} from codewars.com/api/v1"
+                )
+            kata_information = await response.json()
+            return kata_information
+
+    @staticmethod
+    def main_embed(kata_information: dict) -> Embed:
+        """
+        Creates the main embed which displays the description of the kata and the difficulty, along with the name.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        # maps difficulty to color for embed
+        mapping_of_kyu = {
+            8: (221, 219, 218), 7: (221, 219, 218), 6: (236, 182, 19), 5: (236, 182, 19),
+            4: (60, 126, 187), 3: (60, 126, 187), 2: (134, 108, 199), 1: (134, 108, 199)
+        }
+
+        kata_description = kata_information["description"]
+
+        # ensuring it isn't over the length 1024
+        if len(kata_description) > 1024:
+            kata_description = "\n".join(kata_description[:1000].split("\n")[:-1])
+            kata_id = kata_information["id"]
+            kata_description += f"\n[Read more]({f'https://codewars.com/kata/{kata_id}'})"
+
+        # creating the main kata embed
+        kata_embed = Embed(
+            title=kata_information['name'],
+            description=kata_description,
+            color=Color.from_rgb(
+                *mapping_of_kyu[int(kata_information['rank']['name'].replace(' kyu', ''))]
+            ),
+        )
+        kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+        return kata_embed
+
+    @staticmethod
+    def language_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'language embed' which displays all languages the kata supports.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        languages = '\n'.join(map(str.title, kata_information['languages']))
+        language_embed = Embed(
+            title="Languages Supported",
+            description=f"```nim\n{languages}```",
+            color=Colours.python_blue,
+        )
+        return language_embed
+
+    @staticmethod
+    def tags_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'tags embed' which displays all the tags of the Kata.
+
+        Tags are what the Kata is about, and is another word for 'categories'
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
```

Comment:
We understand this from the function head (`def tags_embed(kata_information: dict) -> Embed:`).

Please remove this from all methods.